### DISCUSSION TITLE
Windows support

### DISCUSSION THREAD

**Discussion by ardislu** [paradigmxyz/rivet#3] [Ideas]
Currently, `pnpm install` and `pnpm dev` do not work on Windows because:
- `bun` is a dev dependency (`bun` does not currently support Windows)
- `pnpm dev` sets `NODE_ENV=development`

**Solution 1: update `package.json`**

These issues can be fixed by removing `bun` as a dependency, and using `cross-env` to set the env variable. Here are the required changes to `package.json`:

```diff
diff --git a/package.json b/package.json
index b8dc563..bd94e69 100644
--- a/package.json
+++ b/package.json
@@ -11,7 +11,7 @@
     "build:ds": "vite build -c src/design-system/vite.config.ts",
     "build:inpage": "vite build -c vite.config.inpage.ts",
     "clean": "rimraf dist",
-    "dev": "NODE_ENV=development pnpm build:inpage && vite --open",
+    "dev": "cross-env NODE_ENV=development pnpm build:inpage && vite --open",
     "format": "rome format . --write",
     "lint": "rome check .",
     "lint:fix": "pnpm lint --apply",
@@ -52,9 +52,9 @@
     "@vanilla-extract/sprinkles": "^1.6.0",
     "@vanilla-extract/vite-plugin": "^3.8.0",
     "@vitejs/plugin-react": "^4.0.0-beta.0",
-    "bun": "^0.5.9",
     "chroma-js": "^2.4.2",
     "clsx": "^1.2.1",
+    "cross-env": "^7.0.3",
     "d3-format": "^3.1.0",
     "esbuild": "^0.17.17",
     "esbuild-register": "^3.4.2",
```

After making these changes, I'm able to run the dev server and build the extension on Windows.

**Solution 2: use WSL**

`pnpm install` and `pnpm dev` work as expected in [WSL](https://learn.microsoft.com/en-us/windows/wsl/), without making any changes to the repo.

---

Not sure if removing `bun` breaks anything, it appears unused?

**Comment by jxom** [comment]
Ah think we may have used `bun` in early development. Feel free to make a PR with these changes!

**Reply by ardislu** [reply]
Sounds good, I created #6 with these changes.

**Comment by ardislu** [comment]
Noting this for future reference. 

[Bun v1.0 was released today](https://bun.sh/blog/bun-v1.0) and it includes experimental support for Windows. So this dependency may be re-introduced in the (hopefully near) future once support has stabilized.
